### PR TITLE
add CLI play-playlist option, to play given playlist name.

### PR DESCRIPTION
**Pull request by NicolasToussaint**
Adding a command line option to start playing the playlist given as argument.

I reused the command line argument playlist_name_ from the create_playlist option.

### CODE REVIEW

File: src/core/mainwindow.cpp [hunk 1]
```diff
@@ -2274,6 +2274,13 @@ void MainWindow::CommandlineOptionsReceived(const CommandlineOptions &options) {
     case CommandlineOptions::Player_Next:
       app_->player()->Next();
       break;
+    case CommandlineOptions::Player_PlayPlaylist:
+      if (options.playlist_name().isEmpty()) {
+        qLog(Error) << "ERROR: playlist name missing";
+      } else {
```
Comment (jonaski) [inline src/core/mainwindow.cpp:2280]: }
else {

File: src/core/player.cpp [hunk 2]
```diff
@@ -398,6 +398,38 @@ void Player::NextItem(const Engine::TrackChangeFlags change, const Playlist::Aut
 
 }
 
+void Player::PlayPlaylist(const QString& playlistName) {
+  PlayPlaylistInternal(Engine::Manual, Playlist::AutoScroll_Always, playlistName);
+}
+
+void Player::PlayPlaylistInternal(Engine::TrackChangeFlags change,
```
Comment (jonaski) [inline src/core/player.cpp:405]: One line

File: src/core/player.cpp [hunk 3]
```diff
@@ -398,6 +398,38 @@ void Player::NextItem(const Engine::TrackChangeFlags change, const Playlist::Aut
 
 }
 
+void Player::PlayPlaylist(const QString& playlistName) {
```
Comment (jonaski) [inline src/core/player.cpp:401]: const QString &playlist_name

File: src/core/player.cpp [hunk 4]
```diff
@@ -398,6 +398,38 @@ void Player::NextItem(const Engine::TrackChangeFlags change, const Playlist::Aut
...
+                                  const QString& playlistName) {
+  Playlist* playlist = nullptr;
+  for (Playlist* p : app_->playlist_manager()->GetAllPlaylists()) {
+    if (playlistName == app_->playlist_manager()->GetPlaylistName(p->id())) {
+      playlist = p;
+      break;
+    }
+  }
+
+  if (playlist == NULL) {
```
Comment (jonaski) [inline src/core/player.cpp:416]: nullptr

File: src/core/player.cpp [hunk 5]
```diff
@@ -398,6 +398,38 @@ void Player::NextItem(const Engine::TrackChangeFlags change, const Playlist::Aut
...
 }
 
+void Player::PlayPlaylist(const QString& playlistName) {
+  PlayPlaylistInternal(Engine::Manual, Playlist::AutoScroll_Always, playlistName);
+}
+
+void Player::PlayPlaylistInternal(Engine::TrackChangeFlags change,
+                                  const Playlist::AutoScroll autoscroll,
+                                  const QString& playlistName) {
+  Playlist* playlist = nullptr;
```
Comment (jonaski) [inline src/core/player.cpp:408]: Playlist *playlist = nullptr;

File: src/core/player.cpp [hunk 6]
```diff
@@ -398,6 +398,38 @@ void Player::NextItem(const Engine::TrackChangeFlags change, const Playlist::Aut
 
 }
 
+void Player::PlayPlaylist(const QString& playlistName) {
+  PlayPlaylistInternal(Engine::Manual, Playlist::AutoScroll_Always, playlistName);
+}
+
+void Player::PlayPlaylistInternal(Engine::TrackChangeFlags change,
+                                  const Playlist::AutoScroll autoscroll,
+                                  const QString& playlistName) {
```
Comment (jonaski) [inline src/core/player.cpp:407]: const QString &playlist_name

File: src/core/player.h [hunk 7]
```diff
@@ -195,6 +196,7 @@ class Player : public PlayerInterface {
   void PreviousItem(const Engine::TrackChangeFlags change);
 
   void NextInternal(const Engine::TrackChangeFlags, const Playlist::AutoScroll autoscroll);
+  void PlayPlaylistInternal(Engine::TrackChangeFlags, const Playlist::AutoScroll autoscroll, const QString& playlistName);
```
Comment (jonaski) [inline src/core/player.h:199]: const QString &playlist_name

**Review by jonaski — CHANGES_REQUESTED** [review]

**Comment by jonaski** [comment]
Thanks!

**Comment by NicolasToussaint** [comment]
My pleasure, thanks for the great work !

**Pull request merged**